### PR TITLE
slice helpers with n = 0 & prop = 0 result in empty data frame

### LINKED ISSUE #6184: slice_min now returns all tibble values when prop = 0
On update to dplyr 1.0.8 now when you use slice_min with prop = 0 it returns the whole tibble, it used to return an empty tibble which I think is the intended result by users.

```r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

d <- tibble(a=c(1,2,3,4,5,6,7,8,9,10),
            b=rep("x", 10))

d %>% 
  slice_min(
    order_by = a,
    prop = 0,
    with_ties = FALSE)
#> # A tibble: 10 x 2
#>        a b    
#>    <dbl> <chr>
#>  1     1 x    
#>  2     2 x    
#>  3     3 x    
#>  4     4 x    
#>  5     5 x    
#>  6     6 x    
#>  7     7 x    
#>  8     8 x    
#>  9     9 x    
#> 10    10 x

<sup>Created on 2022-02-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

```

### PR DESCRIPTION
**Pull request by eutwt**
fixes #6184

### CODE REVIEW

**Review by romainfrancois — APPROVED** [review]
LGTM

File: tests/testthat/test-slice.r [hunk 1]
```diff
@@ -245,6 +245,40 @@ test_that("proportion computed correctly", {
...
+  expect_identical(df %>% slice_max(x, n = 0), slice0)
+  expect_identical(df %>% slice_sample(n = 0), slice0)
+
+  expect_identical(df %>% slice_head(prop = 0), slice0)
+  expect_identical(df %>% slice_tail(prop = 0), slice0)
+  expect_identical(df %>% slice_min(x, prop = 0), slice0)
+  expect_identical(df %>% slice_max(x, prop = 0), slice0)
+  expect_identical(df %>% slice_sample(prop = 0), slice0)
+})
+
```
Comment (DavisVaughan) [inline tests/testthat/test-slice.r:281]: Can you please add a NEWS bullet mentioning the closed issue and your GitHub user name? (Follow the pattern of other news bullets if you haven't done this before)

**Review by DavisVaughan — APPROVED** [review]
LGTM, we should probably get the CI working on `main` again (looks like updating rlang snapshots), then rebase this on that to ensure that everything is working as expected

**Comment by eutwt** [comment]
Rebased to tidyverse/main, no other changes

**Pull request closed without merging**